python-sdk/tests_integration/astro_deploy/deploy.sh

```diff
@@ -104,4 +110,49 @@ curl --location --request POST "https://api.astronomer.io/hub/v1" \
                 }
           }"
 
+# Build image and deploy to Single Worker
+BUILD_NUMBER_SINGLE_WORKER=$(awk 'BEGIN {srand(); print srand()}')
+IMAGE_SINGLE_WORKER=${ASTRO_DOCKER_REGISTRY}/${ASTRO_ORGANIZATION_ID}/${ASTRO_DEPLOYMENT_ID_SINGLE_WORKER}:ci-${BUILD_NUMBER_SINGLE_WORKER}
+docker build --platform=linux/amd64 -t "${IMAGE_SINGLE_WORKER}" -f "${SCRIPT_PATH}"/Dockerfile.single_worker "${SCRIPT_PATH}"
+docker login "${ASTRO_DOCKER_REGISTRY}" -u "${ASTRO_KEY_ID_SINGLE_WORKER}" -p "${ASTRO_KEY_SECRET_SINGLE_WORKER}"
+docker push "${IMAGE_SINGLE_WORKER}"
+
+TOKEN_SINGLE_WORKER=$( curl --location --request POST "https://auth.astronomer.io/oauth/token" \
+      --header "content-type: application/json" \
+      --data-raw "{
+          \"client_id\": \"$ASTRO_KEY_ID_SINGLE_WORKER\",
+          \"client_secret\": \"$ASTRO_KEY_SECRET_SINGLE_WORKER\",
+          \"audience\": \"astronomer-ee\",
+          \"grant_type\": \"client_credentials\"}" | jq -r '.access_token' )
+
+# Step 8. Create the Image
+echo "get image id"
+IMAGE_SINGLE_WORKER=$( curl --location --request POST "https://api.astronomer.io/hub/v1" \
+      --header "Authorization: Bearer $TOKEN_SINGLE_WORKER" \
+      --header "Content-Type: application/json" \
+      --data-raw "{
+          \"query\" : \"mutation imageCreate(\n    \$input: ImageCreateInput!\n) {\n    imageCreate (\n    input: \$input\n) {\n    id\n    tag\n    repository\n    digest\n    env\n    labels\n    deploymentId\n  }\n}\",
+          \"variables\" : {
+              \"input\" : {
+                  \"deploymentId\" : \"$ASTRO_DEPLOYMENT_ID_SINGLE_WORKER\",
+                  \"tag\" : \"ci-$BUILD_NUMBER_SINGLE_WORKER\"
+                  }
+              }
+          }" | jq -r '.data.imageCreate.id')
+# Step 9. Deploy the Image
+echo "deploy image for single worker"
+curl --location --request POST "https://api.astronomer.io/hub/v1" \
+        --header "Authorization: Bearer $TOKEN_SINGLE_WORKER" \
+        --header "Content-Type: application/json" \
```

Comment:
reduce redundant code through a function which will work based on a parameter - "single" or "multi"

Reply:
addressed in a3da71dafae23c1217f7b01de0633991809705b7